Update MSYS2 job in GitHub Actions workflow

Foadsf opened the pull request:
- Remove unavailable mingw-w64-x86_64-openmpi package
- Disable MPI in CMake configuration

This change allows the MSYS2 Mingw-w64 build to proceed without
errors, though it disables MPI support. Future work may be needed
to re-enable MPI using an alternative package or build method.

relevant to https://github.com/ElmerCSC/elmerfem/issues/494

Foadsf committed:
Add MSYS2 Mingw-w64 job to GitHub Actions workflow

- Create new 'windows-mingw' job in build.yaml
- Set up MSYS2 environment using msys2/setup-msys2 action
- Install necessary packages including MinGW-w64 GCC, Fortran, CMake
- Configure build with MSYS Makefiles generator
- Build and run quick tests

This change improves Windows support by enabling builds with MSYS2 Mingw-w64,
making Elmer more accessible to Windows users without WSL or admin rights.

relevant to https://github.com/ElmerCSC/elmerfem/issues/494

Foadsf committed:
Update MSYS2 job in GitHub Actions workflow

- Remove unavailable mingw-w64-x86_64-openmpi package
- Disable MPI in CMake configuration

This change allows the MSYS2 Mingw-w64 build to proceed without
errors, though it disables MPI support. Future work may be needed
to re-enable MPI using an alternative package or build method.

relevant to https://github.com/ElmerCSC/elmerfem/issues/494

mmuetzel commented on .github/workflows/build.yaml:
```diff
@@ -94,3 +94,50 @@ jobs:
+            mingw-w64-x86_64-cmake
+            mingw-w64-x86_64-openblas
+            mingw-w64-x86_64-parmetis
+            make
```
It'd probably be a good idea to install the `base-devel` group instead of only `make`.

Foadsf replied:
@mmuetzel thanks for the comment. Do you want to share a patch or send a PR to this specific branch? I think it should be possible to give you write access to this specific branch. But I rather core devs to do that.

mmuetzel replied:
Why can't you make that change yourself? It shouldn't be hard.

Foadsf replied:
@mmuetzel fair point. But I did not explain myself properly. Given your experience it would be great if we could give you write access on some of the MSYS2 / Mingw-w64 / Windows-related branches. Of course, if you are interested.

Foadsf committed:
following this comment https://github.com/ElmerCSC/elmerfem/pull/496#discussion_r1678998232
Improve MSYS2 setup in GitHub Actions workflow

- Replace 'make' with 'base-devel' group in package installation
- This change provides a more comprehensive set of development tools,
  including make and other essential utilities for building

mmuetzel commented on .github/workflows/build.yaml:
```diff
@@ -94,3 +94,53 @@ jobs:
+            -DWITH_Zoltan=OFF \
+            -DWITH_Mumps=OFF \
+            -DCREATE_PKGCONFIG_FILE=ON \
+            -DWITH_MPI=ON
```
I accidentally commented on the commit instead of on the PR. So, repeating here. Sorry for the noise.

``` suggestion
            -DWITH_MPI=ON \
            -DMPIEXEC_EXECUTABLE="C:/Program Files/Microsoft MPI/Bin/mpiexec.exe" \
```

But for this to actually pass in CI, you'd need a new step to install MSMPI.
To install it locally, you can get the installer from here: https://www.microsoft.com/en-us/download/details.aspx?id=105289 (The MS server seems to be down currently.)

I haven't looked into how that could be installed in CI though...

PS: In any case, you are missing a line continuation here.

Foadsf commented:
I hard reset the latest two commits on the `feat/windows-msys2-build` branch, moving them to `fix/fortran-boz-compatibility`. Sorry for the messy try and error folks.

mmuetzel commented:
> I hard reset the latest two commits on the `feat/windows-msys2-build` branch, moving them to `fix/fortran-boz-compatibility`. Sorry for the messy try and error folks.

If you wait for another day or two, you'll no longer need these Fortran flags. MSYS2 should have updated the `mingw-w32-*-msmpi` packages by then. The update is already in their staging repository:
https://packages.msys2.org/queue

mmuetzel commented:
Do you know why this PR triggers the action four times each time you push a change?

![msedge_EUO9i9jKkz](https://github.com/user-attachments/assets/8f9cf9a3-1f47-43e3-9ba1-712522f2bc54)


One is probably because it triggers on a push on your branch, the other one is for the pull request (i.e., for a merge to the base branch).
But why are there two more?

**P.S. by Foad**  follow-up discussions [here](https://discord.com/channels/412182089279209474/697121430395027591/1262769862648791173) on Discord.

raback commented:
Great developmemts! I wonder if this would deserve its own .yaml file? See the comment here: https://github.com/ElmerCSC/elmerfem/pull/487#issuecomment-2260174384

raback commented:
The content of this PR has been manually copied to here:
https://github.com/ElmerCSC/elmerfem/blob/devel/.github/workflows/build-windows-mingw.yaml
Now each platform has its own .yaml file and they may be triggered independently. 

The build action can be seen here:
https://github.com/ElmerCSC/elmerfem/actions/workflows/build-windows-mingw.yaml

raback commented:
Closing this one as the msys2 build is now successfully in the CI.
